clients/drcachesim/tools/schedule_stats.cpp

```diff
@@ -467,17 +467,27 @@ schedule_stats_t::print_counters(const counters_t &counters)
                      "% cpu busy by time, ignoring idle past last instr\n");
     std::cerr << "  Instructions per context switch histogram:\n";
     counters.instrs_per_switch->print();
+    if (!counters.cores_per_thread->empty()) {
+        std::cerr << "  Cores per thread:\n";
+        counters.cores_per_thread->print();
+    }
 }
 
 void
 schedule_stats_t::aggregate_results(counters_t &total)
 {
+    std::unordered_map<workload_tid_t, std::unordered_set<int64_t>, workload_tid_hash_t>
+        cpu_footprint;
     for (const auto &shard : shard_map_) {
         // First update our per-shard data with per-shard stats from the scheduler.
         get_scheduler_stats(shard.second->stream, shard.second->counters);
 
         total += shard.second->counters;
 
+        for (const workload_tid_t wid : shard.second->counters.threads) {
+            cpu_footprint[wid].insert(shard.second->core);
```

Comment:
Would it be more readable and simpler to use `s/shard.second->core/shard.first/`? It would show clearly that the shards have per-core data.

Reply:
Actually those are not always the same: for parallel mode the shard index is used as the hash key, while the stored core is stream->get_output_cpuid() which for as-traced is the original core number and not just the output ordinal.

Reply:
Although these core numbers are just temporary and it's the count we use to construct the histogram: so it doesn't matter for the current usage; but you could imagine someone with a new feature using this temporary cpu_footprint structure for something else: so it still feels better to use the right core identifier.